lib/ae_mdw/activities.ex

```diff
@@ -63,10 +74,58 @@ defmodule AeMdw.Activities do
          {:ok, cursor} <- deserialize_cursor(cursor) do
       {prev_cursor, activities_locators_data, next_cursor} =
         fn direction ->
-          gens_stream = build_gens_stream(state, direction, account_pk, range, cursor)
-          txs_stream = build_txs_stream(state, direction, account_pk, range, cursor)
+          {txi_scope, gen_scope} =
+            case range do
+              {:gen, %Range{first: first_gen, last: last_gen}} ->
+                {
+                  {first_gen, last_gen},
+                  {DbUtil.first_gen_to_txi(state, first_gen, direction),
+                   DbUtil.last_gen_to_txi(state, last_gen, direction)}
+                }
+
+              nil ->
+                {nil, nil}
+            end
+
+          {txi_cursor, local_idx_cursor} =
+            case cursor do
+              {_height, txi, local_idx} -> {txi, local_idx}
+              nil -> {nil, nil}
+            end
+
+          gens_stream = build_gens_stream(state, direction, account_pk, gen_scope, cursor)
+          txs_stream = build_txs_stream(state, direction, account_pk, txi_scope, txi_cursor)
+
+          int_contract_calls_stream =
+            build_int_contract_calls_stream(state, direction, account_pk, txi_scope, cursor)
+
+          ext_contract_calls_stream =
+            build_ext_contract_calls_stream(state, direction, account_pk, txi_scope, cursor)
+
+          stream =
+            [
+              txs_stream,
+              gens_stream,
+              ext_contract_calls_stream,
+              int_contract_calls_stream
+            ]
+            |> Collection.merge(direction)
+            |> Stream.dedup_by(&elem(&1, 0))
```

Comment:
The reason why we need to deduplicate is because some internal transactions may involve the same account, with different positions (e.g. https://mainnet.aeternity.io/mdw/v2/contracts/calls?scope=txi:22942379)